src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -166,6 +171,134 @@ tnsr::ijj<DataType, SpatialDim, Frame> deriv_spatial_metric(
   return deriv_spatial_metric;
 }
 
+template <size_t SpatialDim, typename Frame, typename DataType>
+void time_deriv_of_spatial_metric(
+    const gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*>
+        dt_spatial_metric,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  if (get_size(get<0, 0>(*dt_spatial_metric)) != get_size(get(lapse))) {
+    tnsr::ii<DataType, SpatialDim, Frame> dt_spatial_metric_(get(lapse));
+    *dt_spatial_metric = dt_spatial_metric_;
+  }
+  auto& d0_spatial_metric = *dt_spatial_metric;
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = i; j < SpatialDim; ++j) {
+      d0_spatial_metric.get(i, j) = -get(lapse) * pi.get(i + 1, j + 1);
+      for (size_t k = 0; k < SpatialDim; ++k) {
+        d0_spatial_metric.get(i, j) += shift.get(k) * phi.get(k, i + 1, j + 1);
+      }
+    }
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> time_deriv_of_spatial_metric(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  tnsr::ii<DataType, SpatialDim, Frame> dt_spatial_metric {};
+  GeneralizedHarmonic::time_deriv_of_spatial_metric<SpatialDim, Frame,
+    DataType>(make_not_null(&dt_spatial_metric), lapse, shift, phi, pi);
+  return dt_spatial_metric;
+}
+
+namespace {
+template <size_t SpatialDim, typename Frame, typename DataType>
+struct D4gBuffer;
+
+template <size_t SpatialDim, typename Frame>
+struct D4gBuffer<SpatialDim, Frame, double> {
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+  explicit D4gBuffer(const size_t /*size*/) noexcept {}
+
+  tnsr::ijj<double, SpatialDim, Frame> deriv_of_g;
+  Scalar<double> det_spatial_metric;
+};
+
+template <size_t SpatialDim, typename Frame>
+struct D4gBuffer<SpatialDim, Frame, DataVector> {
```

Comment:
@hrueter and myself also discussed these buffers that @prayush came up with in the context of #1172. As far as I understand things, they accomplish two things: Combine `double` and `DataVector` implementations, and wrap `Variables` to not have to deal with either writing a tag struct for each variables or resort to the `Tags::Temp...` (that I strongly dislike). I quite liked @prayush's approach here, actually. But it would be great to reach a consensus on how we deal with these issues throughout our code.
- I like that with these buffers each variable has a descriptive name, as opposed to something like `Temp<1>`.
- @hrueter and myself notices that if we had the possibility to point into the spatial part of a 4-tensor, we could get rid of some of the tensors in these buffers. For instance, in #1172 there is a `d4_lapse` that we have to fill with its temporal part `d0_lapse` and spatial part `d3_lapse`. All three tensor need to be in the buffer right now, but it would be good to avoid this.

Reply:
I see already that the design of these buffers will require a bit more discussion. I'm not entirely convinced that the temp tags are not the best answer to the problem, especially because the buffers may well be re-used for unrelated quantities, and I think I would prefer a highly general data structure rather than several computation-specific data structures.

So, I've opened an issue #1275, so we can move further discussion there. When you get a chance, please add your input to make sure that your opinion is appropriately represented.